github/GithubIntegration.py

```diff
@@ -213,3 +214,16 @@ def get_app_installation(self, installation_id):
         :rtype: :class:`github.Installation.Installation`
         """
         return self._get_installed_app(url=f"/app/installations/{installation_id}")
+
+    def get_app(self):
+        """
+        :calls: `GET /app <https://docs.github.com/en/rest/reference/apps#get-the-authenticated-app>`_
+        :rtype: :class:`github.GithubApp.GithubApp`
+        """
+
```

Comment:
You could assert in this line

https://github.com/PyGithub/PyGithub/blob/db616376a51789c96afe1e6039f7aecc9b08f56e/github/GithubIntegration.py#L75

that

    assert isinstance(auth, AppAuth), f"GithubIntegration requires github.Auth.AppAuth authentication, not ${type(auth)}"